src/providers/Participant/ParticipantProvider.tsx

```diff
@@ -90,21 +89,19 @@ const ParticipantProvider = ({ children }: Props) => {
   }, [projectModuleList]);
 
   useEffect(() => {
-    if (allProjectModuleList && allProjectModuleList.status === 'success' && allProjectModuleList.data) {
-      const allProjectModules = allProjectModuleList.data;
-      const nextModuleProjects = (participantProjects || []).filter(
-        (project) =>
-          allProjectModules.findIndex(({ id, modules }) => id === project.id && modules && modules.length > 0) !== -1
-      );
+    if (moduleProjectsList && moduleProjectsList.status === 'success' && moduleProjectsList.data) {
+      const nextModuleProjects = moduleProjectsList.data
+        .map((id) => participantProjects.find((project) => project.id === id))
+        .filter((project): project is Project => !!project);
```

Comment:
Is this filter necessary?  Can a project from participant projects not be a project?

Reply:
Good question, I don't believe we're filtering like this elsewhere for projects so it doesn't seem like it should be necessary here.

Reply:
The type will end up being Project | undefined because of the behavior of the ‘find’ method. We do use filter with type predicates in some areas.

Reply:
This ends up with a TS2322 error otherwise: https://terraformationteam.slack.com/archives/C01C69KA23Y/p1712783339082759